thirdparty/DiskANN/include/semaphore.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include <mutex>
+#include <condition_variable>
+
+namespace diskann {
+class Semaphore {
```

Comment:
I'd use the syntax from https://en.cppreference.com/w/cpp/thread/counting_semaphore, so that we could transition the code to C++20 more easily